modules/cluster-logging-collector-log-forward-cloudwatch.adoc

```diff
@@ -0,0 +1,290 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-external.adoc
+
+[id="cluster-logging-collector-log-forward-cloudwatch_{context}"]
+= Forwarding logs to Amazon CloudWatch
+
+You can forward logs to Amazon CloudWatch, a hosted monitoring and log storage service. You can forward logs to CloudWatch in addition to, or instead of, the default OpenShift Logging-managed Elasticsearch log store.
+
+To configure log forwarding to CloudWatch, create a `ClusterLogForwarder` custom resource (CR) with an output for CloudWatch and a pipeline that uses the output.
+
+You can include a specific CloudWatch topic in the output or use the default. The CloudWatch output can use a TCP (insecure) or TLS (secure TCP) connection.
+
+.Procedure
+
+. Create a `Secret` YAML file similar to the following example. Use base64-encoded values for the `username` and `password` fields. For example:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cw-secret
+  namespace: openshift-logging
+data:
+  aws_access_key_id: F1lAUtMIGg2LzANQkVIWVhSRFQKQ
+  aws_secret_access_key: aSEvRkovGEpPMkxZNHhWK1dwb2p0UY1kvHlNdEt6Z0VhNHBSV1J3Lgo=
+----
+
+
+. Create the secret. For example:
++
+[source,terminal]
+----
+$ oc create secret -n openshift-logging cw-secret.yaml
+----
+
+
+. Create a `ClusterLogForwarder` CR YAML file. In it, specify the name of the secret. For example:
++
+[source,yaml]
+----
+apiVersion: "logging.openshift.io/v1"
+kind: ClusterLogForwarder
+metadata:
+  name: instance <1>
+  namespace: openshift-logging <2>
+spec:
+  outputs:
+   - name: cw <3>
+     type: cloudwatch <4>
+     cloudwatch:
+       groupBy: logType <5>
+       region: us-east-2 <6>
+     secret:
+        name: cw-secret <7>
+  pipelines:
+    - name: infra-logs <8>
+      inputRefs: <9>
+        - infrastructure
+        - audit
+        - application
+      outputRefs:
+        - cw <10>
+----
+<1> The name of the `ClusterLogForwarder` CR must be `instance`.
+<2> The namespace for the `ClusterLogForwarder` CR must be `openshift-logging`.
+<3> Specify a name for the output.
+<4> Specify the `cloudwatch` type.
+<5> TBD? Optional: Specify the option CloudWatch uses to group the logs.
+<6> TBD? Optional: Specify the region or regions CloudWatch uses.
```

Comment:
region specifies a single region and is mandatory. groupBy is optional and if omitted defaults to logType.

Reply:
Updated. Thanks. 
@syedriko  Can you confirm whether `region` is optional?

Reply:
Updated. Not optional. Thanks for testing.